Sync Lock: make sure to reinit leader params after scratching ends

ywwg opened the pull request:
Fixes an issue where scratching the leader while another stopped deck was present would cause the leader to change pitch to resync.

daschuer commented:
If you put the renaming commits into a separate PR, they can already merged without extensive testing. We can also ignore the unrelated formatting issues there (or not as you prefer).

daschuer commented:
Oh, I have realized that there are already separate PRs 
This depends on #3944 #3921, #3706

Holzhaus commented on src/engine/sync/enginesync.cpp:
```diff
@@ -381,9 +379,30 @@ void EngineSync::notifyPlayingAudible(Syncable* pSyncable, bool playingAudible)
+        pSyncable->getChannel()->getEngineBuffer()->requestSyncPhase();
+        return;
+    }
+    if (isLeader(pSyncable->getSyncMode())) {
```
You can invert this condition and exit early to reduce nesting depth.

ywwg replied:
done, and added a synchronized check at the top to simplify further

Holzhaus commented on src/engine/sync/synccontrol.cpp:
```diff
@@ -339,21 +343,49 @@ void SyncControl::trackLoaded(TrackPointer pNewTrack) {
+    m_leaderBpmAdjustFactor = kBpmUnity;
+
+    m_pBpmControl->updateLocalBpm();
+    if (isSynchronized()) {
```
You can invert this condition and exit early to reduce nesting depth.

ywwg replied:
done

ywwg commented:
Now that #3944 is in, this one is ready for review!  To demonstrate the issue, play two tracks in sync and then drag the waveform around -- with this change, the scratched waveform snaps back into place. Without, it stays out of sync.

ywwg committed:
Sync Lock: make sure to reinit leader params after scratching ends

Fixes an issue where scratching the leader while another stopped deck was present would cause the leader to change pitch to resync.

ywwg committed:
Sync Lock: Have bpmcontrol override notifySeek

ywwg committed:
Sync Lock: Make sure to update the leader beat distance after a seek

Because of half/double math, a phased seek might cause the Leader to be 0.5 beats off from where it used to be. Therefore after a seek, make sure to update the leader beat distance with the new value.

daschuer reviewed (approved):
LGTM
